lib/modules/code_generator/index.js

```diff
@@ -342,12 +333,8 @@ class CodeGenerator {
     contractCode += "import web3 from 'Embark/web3';\n";
     contractCode += "import EmbarkJS from 'Embark/EmbarkJS';\n";
     contractCode += "let " + contractName + "JSONConfig = " + JSON.stringify(contractJSON) + ";\n";
-    contractCode += "let " + contractName + " = new EmbarkJS.Contract(" + contractName + "JSONConfig);\n";
-
-    contractCode += "\n__embarkContext.execWhenReady(function() {\n";
-    contractCode += "\n" + contractName + ".setProvider(web3.currentProvider);\n";
-    contractCode += "\n" + contractName + ".options.from = web3.eth.defaultAccount;\n";
-    contractCode += "\n});\n";
+    contractCode += contractName + "JSONConfig.web3 = web3;\n";
+    contractCode += "let " + contractName + " = new EmbarkJS.Blockchain.Contract(" + contractName + "JSONConfig);\n";
```

Comment:
You should use String templates, they are easier to use and easier to read.
eg: `let ${contractName} = new EmbarkJS.Blockchain.Contract(${contractName}JSONConfig);\n;`